### ISSUE TITLE
vcsim allows unauthenticated requests against VAPI REST endpoint

### ISSUE THREAD

**Issue by embano1** [vmware/govmomi#1361]
During review of #1360 I realized that it's possible to make unauthenticated requests against the REST API of `vcsim` whereas the SDK endpoint requires authentication. This leads to inconsistent experience when testing against `vcsim` REST API vs. vCenter (which requires REST calls to be authenticated as well).

```bash
# succeeds against vcscim without credentials
GOVC_URL=http://localhost:8989 GOVC_INSECURE=true ./govc tags.ls
embatag1
```

```bash
# fails against vcscim without credentials
GOVC_URL=http://localhost:8989 GOVC_INSECURE=true ./govc ls '/*'
./govc: NotAuthenticated
```

```bash
# fails against VC without credentials
GOVC_URL=https://192.168.178.72 GOVC_INSECURE=true ./govc tags.attached.ls embatag1
./govc: POST https://192.168.178.72/rest/com/vmware/cis/session: 401 Unauthorized
```

**Comment by dougm** [comment]
Right, we have some session related TODOs, I'll see about taking care of this: https://github.com/vmware/govmomi/blob/ffee90f9fba3c9f3d816fcbae0f81f91f3b9b126/vapi/simulator/simulator.go#L207

**Comment by dougm** [comment]
Fixed by #1364
